MedtrumKitUI/ViewModels/Onboarding/DeactivatePatchViewModel.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  DeactivatePatchViewModel.swift
+//  MedtrumKit
+//
+//  Created by Bastiaan Verhaar on 04/04/2025.
+//
+
+import LocalAuthentication
+
+class DeactivatePatchViewModel: ObservableObject {
+    
+    @Published var isDeactivating = false
+    @Published var deactivationError = ""
+    @Published var is300u = false
+    
+    private let nextStep: () -> Void
+    private let pumpManager: MedtrumPumpManager?
+    init(_ pumpManager: MedtrumPumpManager?, _ nextStep: @escaping () -> Void) {
+        self.pumpManager = pumpManager
+        self.nextStep = nextStep
+        
+        guard let pumpManager = self.pumpManager else {
+            return
+        }
+        
+        is300u = pumpManager.state.pumpName.contains("300U")
+    }
+    
+    func deactivate() {
+#if targetEnvironment(simulator)
+        authenticate { success in
+            DispatchQueue.main.async {
+                guard success else {
+                    self.deactivationError = LocalizedString("Authentication failure", comment: "auth failed")
+                    return
+                }
+                
+                self.nextStep()
+            }
+        }
+#else
+        guard let pumpManager = self.pumpManager else {
+            nextStep()
+            return
+        }
+        
+        isDeactivating = true
+        deactivationError = ""
+        
+        authenticate { success in
+            guard success else {
+                DispatchQueue.main.async {
+                    self.deactivationError = LocalizedString("Authentication failure", comment: "auth failed")
+                }
+                return
+            }
+            
+            pumpManager.deactivatePatch { result in
+                DispatchQueue.main.async {
+                    self.isDeactivating = false
+                    
+                    if case .failure(let error) = result {
+                        self.deactivationError = error.localizedDescription
+                        return
+                    }
+                    
+                    self.nextStep()
+                    
+                }
+            }
+        }
+#endif
+    }
+    
+    private func authenticate(success authSuccess: @escaping (Bool) -> Void) {
+        let context = LAContext()
+        var error: NSError?
+        
+        // check whether authentication is possible
+        if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+            // it's possible, so go ahead and use it
+            let reason = "We need to unlock your data."
+            
+            context.evaluatePolicy(.deviceOwnerAuthentication, localizedReason: reason) {  success, _  in
+                authSuccess(success)
+            }
+        } else {
+            // no auth, automatically allow
```

Comment:
? Not sure what the purpose of this is, but not sure if you can allow it like this?

Reply:
The `context.canEvaluatePolicy(.deviceOwnerAuthentication`-function will return false if an iPhone does not have faceID nor touchID and is not a simulator.

So iPhone 5 and older will fall in this category.
Just added to be complete and prevent app crashes